core/src/main/java/org/bitcoinj/base/Coin.java

```diff
@@ -354,6 +372,27 @@ public BigDecimal toBtc() {
         return satoshiToBtc(this.value);
     }
 
+    /**
+     * Write the amount into the given buffer as 8 bytes in little-endian order.
+     *
+     * @param buf buffer to write into
+     * @return the buffer
+     * @throws BufferOverflowException if the value doesn't fit the remaining buffer
+     */
+    public ByteBuffer write(ByteBuffer buf) throws BufferOverflowException {
+        return buf.order(ByteOrder.LITTLE_ENDIAN).putLong(this.value);
+    }
+
+    /**
+     * Allocates a byte array and serializes the amount.
+     *
+     * @return serialized amount
+     */
+    public byte[] serialize() {
+        ByteBuffer buf = ByteBuffer.allocate(BYTES);
+        return write(buf).array();
+    }
+
```

Comment:
I still think that `serialize()` should not be using an intermediate `ByteBuffer` here, but I suppose you could call that "premature optimization", so I'm going to go ahead an approve the PR anyway.

Reply:
I don't see a way to prevent this, if we want to use `ByteBuffer.putLong(value)` and *not* do our own bitshuffling.